### DISCUSSION TITLE
Provide new_relic integration

### DISCUSSION THREAD

**Discussion by paneq** [RailsEventStore/rails_event_store#1167] [Ideas]
provide ruby_event_store-new_relic integration

```
RailsEventStoreActiveRecord::EventRepository.class_eval do
  include ::NewRelic::Agent::MethodTracer

  add_method_tracer :append_to_stream
  add_method_tracer :link_to_stream
end if defined?(RailsEventStoreActiveRecord::EventRepository)

RubyEventStore::Mappers::Default.class_eval do
  include ::NewRelic::Agent::MethodTracer

  add_method_tracer :event_to_serialized_record
  add_method_tracer :serialized_record_to_event
end


Publisher.class_eval do # aka AggregateRoot
  include ::NewRelic::Agent::MethodTracer

  add_method_tracer :apply
end
```

**Comment by paneq** [comment]
Some questions:

* should we do it automatically upon detecting new relic?
* have it per every gem
  * `require 'ruby_event_store/new_relic` manually in initializer
  * `require 'rails_event_store_active_record/new_relic` manually in initializer
  * etc ?
* have a separate gem such as `ruby_event_store-new_relic` with all integrations?

**Comment by mostlyobvious** [comment]
Could that work through `AS::Notifications`?

**Comment by paneq** [comment]
@pawelpacana interesting idea. related RailsEventStore/rails_event_store#244

**Comment by swistak35** [comment]
Related links:

* https://docs.newrelic.com/docs/agents/ruby-agent/api-guides/ruby-custom-instrumentation
* https://github.com/newrelic/extends_newrelic_rpm
* https://github.com/newrelic/rpm/blob/master/lib/new_relic/agent/instrumentation/action_view_subscriber.rb (newrelic use `ActiveSupport::Notifications` to instrument some parts of rails)
